src/main/kotlin/controllers/CacheChooserController.kt

```diff
@@ -281,53 +306,169 @@ class CacheChooserController(
+                localDate.format(DateTimeFormatter.ISO_LOCAL_DATE)
+            } catch (e: Exception) {
+                null
+            }
+        } ?: "unknown"
+        
+        val buildStr = if (cache.builds.isNotEmpty()) {
+            val build = cache.builds.first()
+            if (build.minor != null) {
+                "rev${build.major}.${build.minor}"
+            } else {
+                "rev${build.major}"
+            }
+        } else {
+            "revunknown"
+        }
+        
+        val destFolderName = "$dateStr-$buildStr"
+        val destFolder = File("${AppConstants.CACHES_DIRECTORY}/$destFolderName")
 
         Thread {
             try {
-                val conn = URL("$RUNESTATS_URL/$cacheName").openConnection()
+                // Check if disk.zip is available (disk_store_valid should be true for new engine caches)
+                // Download from OpenRS2 disk.zip endpoint which provides .dat2/.idx files directly
+                val url = "$OPENRS2_URL/caches/${cache.scope}/${cache.id}/disk.zip"
+                val conn = URL(url).openConnection()
                 conn.addRequestProperty("User-Agent", "osrs-environment-exporter")
                 BufferedInputStream(conn.getInputStream()).use { inputStream ->
-                    val tarIn = TarArchiveInputStream(
-                        GzipCompressorInputStream(inputStream)
-                    )
-                    var tarEntry: TarArchiveEntry? = tarIn.nextTarEntry
-                    while (tarEntry != null) {
-                        val dest = File(destFolder, tarEntry.name)
-                        if (tarEntry.isDirectory) {
-                            dest.mkdirs()
+                    // Delete existing cache folder if it exists to avoid conflicts
+                    val cacheDir = File(destFolder, "cache")
+                    if (cacheDir.exists()) {
+                        cacheDir.deleteRecursively()
+                    }
+                    
+                    val zipIn = ZipInputStream(inputStream)
+                    var zipEntry: ZipEntry? = zipIn.nextEntry
+                    while (zipEntry != null) {
+                        // OpenRS2 disk.zip format: cache/<filename>.dat2 or cache/<filename>.idx
+                        // Extract directly to destFolder/cache/<filename>
+                        if (!zipEntry.isDirectory) {
+                            val entryName = zipEntry.name
+                            // Remove the leading "cache/" prefix from the path if present
+                            val relativePath = if (entryName.startsWith("cache/")) {
+                                entryName.substring(6) // Remove "cache/" prefix
+                            } else {
+                                entryName
+                            }
+                            val dest = File(destFolder, "cache/$relativePath")
+                            dest.parentFile?.mkdirs()
```

Comment:
Have `File(destFolder, "cache")` in another variable, you'll avoid the need for null-checking parentFile and also avoid the need for manual path joining here

Reply:
Since `relativePath` may contain a slash (subdirectories), perhaps this is fine as-is